src/main/java/org/broadinstitute/hellbender/utils/variant/writers/IntervalFilteringVcfWriter.java

```diff
@@ -117,29 +131,32 @@ public IntervalFilteringVcfWriter(final VariantContextWriter writer, final List<
         Utils.nonEmpty(intervals);
         Utils.nonNull(mode);
 
-        this.writer = writer;
+        this.underlyingWriter = writer;
         this.detector = OverlapDetector.create(intervals);
         this.mode = mode;
     }
 
     @Override
     public void writeHeader(final VCFHeader header) {
-        writer.writeHeader(header);
+        underlyingWriter.writeHeader(header);
     }
 
     @Override
     public void setHeader(final VCFHeader header) {
-        writer.setHeader(header);
+        underlyingWriter.setHeader(header);
     }
 
     @Override
     public void close() {
-        writer.close();
+        underlyingWriter.close();
+        if (filteredCount > 0) {
```

Comment:
I would probably move the output ahead of the close() since if there's an exception in the tool there's often an exception in the closing it but you might still want that information to debub.

Reply:
Fair enough